src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -13,8 +13,14 @@ namespace AutoRest.Swagger.Model.Utilities
 {
     public static class ValidationUtilities
     {
-        private static readonly string XmsPageable = "x-ms-pageable";
+        private const string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private const string AzureHostSignature = "management.azure.com";
+
+        // if x-ms-parameterized-host extension is used or if the host is anything of the form management.azure, assume this to be an ARMServiceDefinition
+        // TODO: refine this definition for azure data plain also check x-ms-parameterized-host in detail to ensure this is ARM
+        public static bool IsARMServiceDefinition(ServiceDefinition serviceDefinition) => 
```

Comment:
long run yes, but indeed this is currently double effort (now, only who asks for validation will get validation, so no, PetStore will no longer get flagged) - also I think validation categories (ARM, non-ARM) should be triggerable from outside, not by looking at the Swagger!!!